src/main/java/edu/unc/lib/boxc/migration/cdm/validators/PermissionsValidator.java

```diff
@@ -0,0 +1,118 @@
+package edu.unc.lib.boxc.migration.cdm.validators;
+
+import edu.unc.lib.boxc.auth.api.UserRole;
+import edu.unc.lib.boxc.migration.cdm.exceptions.MigrationException;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+import edu.unc.lib.boxc.migration.cdm.model.PermissionsInfo;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PermissionsValidator {
+    private MigrationProject project;
+    private List<String> patronRoles;
+
+    /**
+     * Validate the permissions mappings for this project.
+     * @param force if true, incomplete, overlapping and duplicate mappings will be ignored
+     * @return A list of errors. An empty list will be returned for a valid mapping
+     */
+    public List<String> validateMappings(boolean force) {
+        List<String> errors = new ArrayList<>();
+        Path path = project.getPermissionsPath();
+        String defaultEveryone = null;
+        String defaultAuthenticated = null;
+
+        try (
+            Reader reader = Files.newBufferedReader(path);
+            CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT
+                    .withFirstRecordAsHeader()
+                    .withHeader(PermissionsInfo.CSV_HEADERS)
+                    .withTrim());
+        ) {
+            int i = 2;
+            for (CSVRecord csvRecord : csvParser) {
+                if (csvRecord.size() != 3) {
+                    errors.add("Invalid entry at line " + i + ", must be 3 columns but were " + csvRecord.size());
+                    i++;
+                    continue;
+                }
+                String id = csvRecord.get(0);
+                String everyone = csvRecord.get(1);
+                String authenticated = csvRecord.get(2);
+
+                // default values
+                if (PermissionsInfo.DEFAULT_ID.equals(id)) {
+                    if (defaultEveryone != null && defaultAuthenticated != null) {
+                        errors.add("Can only map default permissions once, encountered reassignment at line " + i);
+                    } else {
+                        defaultEveryone = everyone;
+                        defaultAuthenticated = authenticated;
```

Comment:
It seems like these values are just used for determining if there are multiple default ids. It would be clearer to store a single boolean instead ("hasDefault" or something like that)